azure-storage-blob/src/Blob/BlobRestProxy.php

```diff
@@ -250,6 +258,26 @@ public function setSingleBlobUploadThresholdInBytes($val)
         $this->blockSize = $val > $this->blockSize ? $this->blockSize : $val;
     }
 
+    /**
+     * Set the bearer token for use with OAuth authentication
+     *
+     * @param string $val the token to be included in the authorization header.
+     *
+     * @return void
+     */
+    public function setBearerToken($val)
+    {
+        if (preg_match("/Bearer\s((.*)\.(.*)\.(.*))/", $val, $matches)) {
```

Comment:
Should put the this logic into `OAuthAuthScheme.php`. `OAuthAuthScheme` constructor should only accept bear token value, instead of whole header value.